tools/msys2_setup_buildenv_stage2_32_64.sh

```diff
@@ -0,0 +1,211 @@
+
+  run_step "Upgrading pip" "python -m pip install -U pip"
+  run_step "Installing Python clang" "python -m pip install clang"
+  run_step "Installing Python setuptools and wheel" "python -m pip install setuptools wheel"
+  run_step "Installing Python jinja2" "python -m pip install jinja2"
+
+  # Nov 2023
+  # From https://github.com/miurahr/aqtinstall Install section
+  # When you want to use it on MSYS2/Mingw64 environment, you need to set environmental variable export SETUPTOOLS_USE_DISTUTILS=stdlib,
+  # because of setuptools package on mingw wrongly raise error VC6.0 is not supported
+  export SETUPTOOLS_USE_DISTUTILS=stdlib
+
+  # this will also install dependencies and see note above re psutil issue
+  run_step "Installing Python package aqtinstall" "python -m pip install aqtinstall"
+fi
+
+if [[ $INSTALL_QT -eq 1 ]]; then
+  # if modules use syntax -m MODULE [MODULE]
+  QT_MODULES=
+  # supported versions of dependencies
+  QT_TOOLS=
+  QT_HOST=windows
+  QT_TARGET=desktop
+  QT_ARCH=
+  QT_INSTALL_DIR=qt
+
+  new_step "Searching for available Qt ${QT_VERSION} architectures"
+
+  qtarchlist="$(python -m aqt list-qt ${QT_HOST} ${QT_TARGET} --arch ${QT_VERSION})"
+
+  IFS=' ' read -ra qtarcharray <<< ${qtarchlist}
+
+  for arch in "${qtarcharray[@]}"; do
+    if [[ "${arch}" == win${MSYSTEM:(-2)}_mingw* ]];then
+      QT_ARCH=${arch}
+      break
+    fi
+  done
+
+  if [[ -z "$QT_ARCH" ]]; then
+    end_step 1 "Unable to find suitable Qt architecture in Qt ${QT_VERSION} for ${MSYSTEM}"
+  else
+    end_step 0 ""
+  fi
+
+  run_step "Installing Qt ${QT_VERSION}" \
+  "python -m aqt install-qt --outputdir ${QT_INSTALL_DIR} ${QT_HOST} ${QT_TARGET} ${QT_VERSION} ${QT_ARCH} ${QT_MODULES}"
+fi
+
+if [[ $DOWNLOAD_ARM -eq 1 ]]; then
+  # Note: the version needs to be kept in sync with EdgeTX/build-edgetx/dev/Dockerfile
+
+  DOWNLOAD_FILE="gcc-arm-none-eabi-10-2020-q4-major-win32.exe"
+
+  run_step "Downloading ARM installer" \
+  "wget -c -O ${DOWNLOAD_DIR}/${DOWNLOAD_FILE} --progress=bar:force:noscroll --no-check-certificate \
+  'https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/${DOWNLOAD_FILE}?revision=9a4bce5a-7577-4b4f-910d-4585f55d35e8&rev=9a4bce5a75774b4f910d4585f55d35e8&hash=9770A44FEA9E9CDAC0DD9A009190CC8B'"
+fi
+
+echo "This stage of setting up EdgeTX build environment has finished"
+
+if [[ $DOWNLOAD_ARM -eq 1 ]]; then
+  echo "
+  Next steps:
+  1. Exit this terminal session
+
+  2. In Windows, install the downloaded ARM GNU 11.3.1 toolchain
+
+    Launch the installer ${DOWNLOAD_FILE}
+    - Accept the default installation folder and press Next
+    - Select Add path to environment variable to the list of options and press Finish
+
+  3. Allow MSYS2 to inherit the Windows path to be able to access the arm-none-eabi toolchain
+
+    Edit the file <MSYS2 install directory>\mingw64.ini eg C:\msys64\mingw64.ini
+    - Uncomment the line #MSYS2_PATH_TYPE=inherit
+    - Save the file
+  "
```

Comment:
IMHO better, than setting an operating system wide setting for EdgeTX, would be to explicitly use `-DARM_TOOLCHAIN_DIR` in EdgeTX CMake line, e.g. here then with ARM GNU  11.3.1 `-DARM_TOOLCHAIN_DIR="C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\11.3 rel1\bin"`.

Setting something system wide might affect users building other (non EdgeTX) code on their systems.

Reply:
Apparently this does not work anyway as there are spaces in the path.
Let's just say there was robust discussion when I posted these new scripts about how best to pass the ARM path.
I too prefer the -D method.